server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java

```diff
@@ -415,6 +420,39 @@ OnDiskState loadBestOnDiskState(boolean checkClean) throws IOException {
                     }
 
                     try (DirectoryReader directoryReader = DirectoryReader.open(directory)) {
+
+                        if (logger.isDebugEnabled()) {
+                            final var segmentsFileName = directoryReader.getIndexCommit().getSegmentsFileName();
```

Comment:
In the context of the recent issue I had to troubleshoot, logging the `IndexCommit`'s user data map would have been very helpful. Can we add those too?

Reply:
Good point, added in 3af38cf292d.